proto/engine/v1/execution_engine.go

```diff
@@ -0,0 +1,92 @@
+package enginev1
+
+import "github.com/prysmaticlabs/prysm/v5/encoding/bytesutil"
+
+type Cloneable[T any] interface {
+	Copy() T
+}
```

Comment:
I'm going to approve so you can move on, but my nitpick is that we should name the interface after the method (`Cloneable` does not follow intuitively from `Copy` - the reflexive naming here should be something more like `Copyer`). If you want to address that now I'd appreciate it and give a fast follow-up approval, or you can do it in the follow-up PRs. Also, should the interface be exported?